src/tests/intg/test_files_provider.py

```diff
@@ -0,0 +1,689 @@
+
+
+def test_getgrnam_members_users_multiple(setup_pw_with_canary,
+                                         setup_gr_with_canary,
+                                         files_domain_only):
+    """
+    Multiple users are linked with a group
+    """
+    user_and_group_setup(setup_pw_with_canary,
+                         setup_gr_with_canary,
+                         [user1, user2],
+                         [group12],
+                         False)
+    members_check([group12])
+
+
+def test_getgrnam_members_groups_first(setup_pw_with_canary,
+                                       setup_gr_with_canary,
+                                       files_domain_only):
+    """
+    A group is linked with a user
+    """
+    user_and_group_setup(setup_pw_with_canary,
+                         setup_gr_with_canary,
+                         [user1],
+                         [group1],
+                         True)
+    members_check([group1])
+
+
+def test_getgrnam_ghost(setup_pw_with_canary,
+                        setup_gr_with_canary,
+                        files_domain_only):
+    """
+    Test that a group with members while the members are not present
+    are added as ghosts. This is also what nss_files does, getgrnam would
+    return group members that do not exist as well.
+    """
+    user_and_group_setup(setup_pw_with_canary,
+                         setup_gr_with_canary,
+                         [],
+                         [group12],
+                         False)
+    check_group(group12)
+    for member in group12['mem']:
+        res, _ = call_sssd_getpwnam(member)
+        assert res == NssReturnCode.NOTFOUND
+
+
+def ghost_and_member_test(pw_ops, grp_ops, reverse):
+    user_and_group_setup(pw_ops,
+                         grp_ops,
+                         [user1],
+                         [group12],
+                         False)
+    check_group(group12)
+
+    # We checked that the group added has the same members as group12,
+    # so both user1 and user2. Now check that user1 is a member of
+    # group12 and its own primary GID but user2 doesn't exist, it's
+    # just a ghost entry
+    res, groups = sssd_id_sync('user1')
+    assert res == sssd_id.NssReturnCode.SUCCESS
+    assert len(groups) == 2
+    assert 'group12' in groups
+
+    res, _ = call_sssd_getpwnam('user2')
+    assert res == NssReturnCode.NOTFOUND
+
+
+def test_getgrnam_user_member_and_ghost(setup_pw_with_canary,
+                                        setup_gr_with_canary,
+                                        files_domain_only):
+    """
+    Test that a group with one member and one ghost.
+    """
+    ghost_and_member_test(setup_pw_with_canary,
+                          setup_gr_with_canary,
+                          False)
+
+
+def test_getgrnam_user_member_and_ghost(setup_pw_with_canary,
+                                        setup_gr_with_canary,
+                                        files_domain_only):
```

Comment:
fixed, the reverse still makes sense because it makes the files provider either add the users first and then link the groups, creating member/memberof pairs or makes the files provider create the groups first with ghost entries and then convert them to proper pairs.